code/game/machinery/vending.dm

```diff
@@ -97,6 +99,14 @@
 	
 	var/can_move = 1	//if you can wrench the machine out of place
 
+	component_types = list(

```

Comment:
Why are vendors upgradable? The delays are for the animations on some types, it shouldn't change.